pkg/miekg/miekg_test.go

```diff
@@ -1,13 +1,483 @@
 package miekg
 
 import (
-	"github.com/zmap/dns"
-	"gotest.tools/v3/assert"
 	"net"
+	"reflect"
 	"regexp"
+	"strconv"
 	"testing"
+
+	"github.com/zmap/dns"
+	"github.com/zmap/zdns/pkg/zdns"
+	"gotest.tools/v3/assert"
 )
 
+var mockResults = make(map[string]Result)
+
+var status = zdns.STATUS_NOERROR
+
+type MockLookupClient struct{}
+
+func (mc MockLookupClient) ProtocolLookup(s *Lookup, q Question, nameServer string) (interface{}, zdns.Trace, zdns.Status, error) {
+	if res, ok := mockResults[q.Name]; ok {
+		return res, nil, status, nil
+	} else {
+		return nil, nil, zdns.STATUS_NXDOMAIN, nil
+	}
+}
+
+func InitTest(t *testing.T) (*zdns.GlobalConf, Lookup, MockLookupClient) {
+	gc := new(zdns.GlobalConf)
+	gc.NameServers = []string{"127.0.0.1"}
+
+	glf := new(GlobalLookupFactory)
+	glf.GlobalConf = gc
+
+	rlf := new(RoutineLookupFactory)
+	rlf.Factory = glf
+	rlf.Client = new(dns.Client)
+
+	l, err := rlf.MakeLookup()
+	if l == nil || err != nil {
+		t.Error("Failed to initialize lookup")
+	}
+
+	a := Lookup{Factory: rlf}
+	mc := MockLookupClient{}
+
+	return gc, a, mc
+}
+
+func TestOneA(t *testing.T) {
+	gc, a, mc := InitTest(t)
+	mockResults["example.com"] = Result{
+		Answers: []interface{}{Answer{
+			Ttl:    3600,
+			Type:   "A",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "192.0.2.1",
+		}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       DNSFlags{},
+	}
+	res, _, _, _ := a.DoTargetedLookup(mc, "example.com", gc.NameServers[0], true, false)
+	verifyResult(t, res.(IpResult), []string{"192.0.2.1"}, nil)
+}
+
+func TestTwoA(t *testing.T) {
+	gc, a, mc := InitTest(t)
+	mockResults["example.com"] = Result{
+		Answers: []interface{}{Answer{
+			Ttl:    3600,
+			Type:   "A",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "192.0.2.1",
+		},
+			Answer{
+				Ttl:    3600,
+				Type:   "A",
+				Class:  "IN",
+				Name:   "example.com",
+				Answer: "192.0.2.2",
+			}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       DNSFlags{},
+	}
+	res, _, _, _ := a.DoTargetedLookup(mc, "example.com", gc.NameServers[0], true, false)
+	verifyResult(t, res.(IpResult), []string{"192.0.2.1", "192.0.2.2"}, nil)
+}
+
+func TestQuadAWithoutFlag(t *testing.T) {
```

Comment:
What is this function trying to test? 

Also, generally could we add a line to each of the test cases as to the goal of the UT?

Reply:
Added comments